Sources/SKCore/BuildSystemManager.swift

```diff
@@ -334,3 +299,55 @@ extension BuildSystemManager {
     return self.watchedFiles[uri]?.mainFile
   }
 }
+
+// MARK: - Build settings logger
+
+/// Shared logger that only logs build settings for a file once unless they change
+fileprivate actor BuildSettingsLogger {
+  static let shared = BuildSettingsLogger()
+
+  private var loggedSettings: [DocumentURI: FileBuildSettings] = [:]
```

Comment:
I know we likely already keep a document -> settings in memory for all the workspaces, but it would be nice to not (potentially) double it. Are you worried about some race if this was removed on close?

Reply:
AFAIK we don’t keep that mapping around anywhere. We used to but I removed the caches in `SwiftLanguageServer` and `ClangLanguageServer`. 

`SwiftPMWorkspace`, for example, gets the build settings from the build graph and doesn’t have any file -> build settings cache.

Reply:
I’m merging this now. If you think this should be implemented otherwise, I can address it in a follow-up PR.